### PR TITLE
Don't add Frame Rate, Audio, Subtitles to IFrame tag

### PR DESCRIPTION
**Pull request by koln67**
HLS Spec 4.3.4.3:

 #EXT-X-I-FRAME-STREAM-INF:<attribute-list>

 All attributes defined for the EXT-X-STREAM-INF tag (Section 4.3.4.2)
 are also defined for the EXT-X-I-FRAME-STREAM-INF tag, except for the
 FRAME-RATE, AUDIO, SUBTITLES, and CLOSED-CAPTIONS attributes.

### CODE REVIEW

File: packager/hls/base/master_playlist.cc [hunk 1]
```diff
@@ -218,28 +218,34 @@ void BuildStreamInfTag(const MediaPlaylist& playlist,
                     variant.text_codecs.end());
   tag.AddQuotedString("CODECS", base::JoinString(all_codecs, ","));
 
+  const std::string I_Frame_Tag("#EXT-X-I-FRAME-STREAM-INF");
```
Comment (kqyang) [inline packager/hls/base/master_playlist.cc:221]: Define a boolean variable instead:

const bool is_iframe_playlist = playlist.stream_type() == MediaPlaylist::MediaPlaylistStreamType::kVideoIFramesOnly;

File: packager/hls/base/master_playlist.cc [hunk 2]
```diff
@@ -218,28 +218,34 @@ void BuildStreamInfTag(const MediaPlaylist& playlist,
...
 
     const std::string video_range = playlist.GetVideoRange();
     if (!video_range.empty())
       tag.AddString("VIDEO-RANGE", video_range);
   }
 
-  if (variant.audio_group_id) {
-    tag.AddQuotedString("AUDIO", *variant.audio_group_id);
-  }
+  if (tag_name.compare(I_Frame_Tag) != 0) {
```
Comment (kqyang) [inline packager/hls/base/master_playlist.cc:241]: This change shouldn't be needed variant.audio_group_id and variant.text_group_id are null for iframe only playlist.

File: packager/hls/base/master_playlist.cc [hunk 3]
```diff
@@ -218,28 +218,34 @@ void BuildStreamInfTag(const MediaPlaylist& playlist,
...
     tag.AddNumberPair("RESOLUTION", width, 'x', height);
 
     // Right now the frame-rate returned may not be accurate in some scenarios.
     // TODO(kqyang): Fix frame-rate computation.
-    const double frame_rate = playlist.GetFrameRate();
-    if (frame_rate > 0)
+    if (tag_name.compare(I_Frame_Tag) != 0) {
+      const double frame_rate = playlist.GetFrameRate();
+      if (frame_rate > 0)
       tag.AddFloat("FRAME-RATE", frame_rate);
```
Comment (kqyang) [inline packager/hls/base/master_playlist.cc:233]: 2 spaces indent.

File: packager/hls/base/master_playlist.cc (hunk 1 above)
Comment (koln67) [reply packager/hls/base/master_playlist.cc]: Changed

File: packager/hls/base/master_playlist.cc (hunk 2 above)
Comment (koln67) [reply packager/hls/base/master_playlist.cc]: Change removed

File: packager/hls/base/master_playlist.cc (hunk 3 above)
Comment (koln67) [reply packager/hls/base/master_playlist.cc]: Indented

**Comment by koln67** [comment]
In the latest commit, the golden files were updated, and one of the output m3u8 files has no FRAMERATE attribute in the IFRAME tag, as expected.

File: packager/app/test/testdata/audio-video-with-codec-switching-encryption-trick-play/output.mpd [hunk 4]
```diff
@@ -2,28 +2,14 @@
 <!--Generated with https://github.com/google/shaka-packager version <tag>-<hash>-<test>-->
```
Comment (kqyang) [inline packager/app/test/testdata/audio-video-with-codec-switching-encryption-trick-play/output.mpd:2]: Can you revert this file?

I know it was changed by running the script. Unfortunately, it is not deterministic right now..

File: packager/app/test/testdata/audio-video-with-codec-switching/output.mpd [hunk 5]
```diff
@@ -2,32 +2,32 @@
 <!--Generated with https://github.com/google/shaka-packager version <tag>-<hash>-<test>-->
```
Comment (kqyang) [inline packager/app/test/testdata/audio-video-with-codec-switching/output.mpd:2]: ditto

File: packager/app/test/testdata/live-profile-and-encryption-and-mult-files/output.mpd [hunk 6]
```diff
@@ -7,27 +7,27 @@
       <ContentProtection schemeIdUri="urn:uuid:1077efec-c0b2-4d02-ace3-3c1e52e2fb4b">
         <cenc:pssh>AAAANHBzc2gBAAAAEHfv7MCyTQKs4zweUuL7SwAAAAExMjM0NTY3ODkwMTIzNDU2AAAAAA==</cenc:pssh>
```
Comment (kqyang) [inline packager/app/test/testdata/live-profile-and-encryption-and-mult-files/output.mpd:8]: ditto

File: packager/hls/base/master_playlist.cc [hunk 7]
```diff
@@ -218,16 +218,21 @@ void BuildStreamInfTag(const MediaPlaylist& playlist,
                     variant.text_codecs.end());
   tag.AddQuotedString("CODECS", base::JoinString(all_codecs, ","));
 
+  const bool is_iframe_playlist = playlist.stream_type() ==
```
Comment (kqyang) [inline packager/hls/base/master_playlist.cc:221]: Please move it before line 231.

File: packager/hls/base/master_playlist.cc [hunk 8]
```diff
@@ -218,16 +218,21 @@ void BuildStreamInfTag(const MediaPlaylist& playlist,
...
+  const bool is_iframe_playlist = playlist.stream_type() == 
+             MediaPlaylist::MediaPlaylistStreamType::kVideoIFramesOnly;
+
   uint32_t width;
   uint32_t height;
   if (playlist.GetDisplayResolution(&width, &height)) {
     tag.AddNumberPair("RESOLUTION", width, 'x', height);
 
     // Right now the frame-rate returned may not be accurate in some scenarios.
     // TODO(kqyang): Fix frame-rate computation.
```
Comment (kqyang) [inline packager/hls/base/master_playlist.cc:230]: Move the two lines inside the "if".

File: packager/hls/base/master_playlist.cc (hunk 7 above)
Comment (koln67) [reply packager/hls/base/master_playlist.cc]: Moved

File: packager/hls/base/master_playlist.cc (hunk 8 above)
Comment (koln67) [reply packager/hls/base/master_playlist.cc:227]: Moved, same as earlier comment

File: packager/app/test/testdata/live-profile-and-encryption-and-mult-files/output.mpd (hunk 6 above)
Comment (koln67) [reply packager/app/test/testdata/live-profile-and-encryption-and-mult-files/output.mpd]: Reverted

File: packager/app/test/testdata/audio-video-with-codec-switching/output.mpd (hunk 5 above)
Comment (koln67) [reply packager/app/test/testdata/audio-video-with-codec-switching/output.mpd]: Reverted

File: packager/app/test/testdata/audio-video-with-codec-switching-encryption-trick-play/output.mpd (hunk 4 above)
Comment (koln67) [reply packager/app/test/testdata/audio-video-with-codec-switching-encryption-trick-play/output.mpd]: Reverted

**Pull request merged**